client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcConnection.java

```diff
@@ -405,6 +427,62 @@ public void testSessionProperties()
         }
     }
 
+    @Test(timeOut = 60_000)
+    public void testCancellationOnStatementClose()
```

Comment:
Please add a non-threaded test where ResultSet is being processed, but not consumed fully and not closed.
You can use blackhole again for this, with no processing delay, but with "infinite" number of pages instead.

Please add sth like "Concurrent" or "Threaded", or "Async" to method names of tests testing concurrent close.